### PR TITLE
make sure that no one calls decay for a goldstone

### PR DESCRIPTION
**Pull request by wkotlarski**
Hi @Expander 

IMO this PR should go before others we are now discussing (#409, #410). Here's what was fishy. In an autogenerated code we made sure that we do not generate decays of Goldstone bosons. But you can still call them by hand, as was done in some test. This PR makes sure it cannot happen. If you try this, you'll get something like

```
unknown location(0): fatal error: in "test_MRSSM2_FlexibleDecay": flexiblesusy::OutOfBoundsError: Error in Ah(0)->{barFd(2),Fd(2)} decay. Decaying particle is a Goldstone.
```

I have implemented a simplest way to get the number of Goldstones in a multiplet but I'm happy to change it if you want it.

PS. I'm not 100% sure how it was related to the fact that after change of SM mW some Ah decays became 0. Only charged Goldstone should be controlled directly by mW.

### CODE REVIEW

**Review by Expander — APPROVED** [review]

**Comment by Expander** [comment]
Perfect, many thanks! I have no further comments.

**Comment by Expander** [comment]
Please feel free to merge this PR when you think it is ready and the tests pass.

**Comment by wkotlarski** [comment]
Thanks! Once tests pass, I'll merge it to `devepment` and then `development` to `feature-MW-tweaks`. Then I'll fix remaining tests.

**Pull request merged**
